pandas/core/frame.py

```diff
@@ -4136,7 +4194,7 @@ def _maybe_cache_changed(self, item, value: Series, inplace: bool) -> None:
     # ----------------------------------------------------------------------
     # Unsorted
 
-    def query(self, expr: str, inplace: bool = False, **kwargs):
+    def query(self, expr: str, inplace: bool = False, **kwargs) -> None:
```

Comment:
Probably needs an overload depending on `inplace` similar to others in this pr?

Reply:
For some reason, I cannot add the deprecation decorated:

```py
@deprecate_nonkeyword_arguments(version=None, allowed_args=["self", "expr"])
def query(self, expr: str, inplace: bool = False, **kwargs) -> DataFrame | None:
```

I think the reason is that it `expr` refers to local variables which are later looked up. Adding the decorator messes that somehow up.